### PR TITLE
Use string instead of enum for xds resource type

### PR DESCRIPTION
**Pull request by anicr7**
Currently the xdsresource interface defines a function to get an enum associated with the resource. But as the enum is defined in xdsresource package it is not easy to extend it for custom xds resources.

The type enum is mostly utilized for coming up a version agnostic name for a type and so using a string is cleaner and allows custom xds resources to specify their own names. 

cc: @easwars 

RELEASE NOTES: none

### CODE REVIEW

File: xds/internal/xdsclient/authority.go
Comment (easwars) [inline xds/internal/xdsclient/authority.go:332]: Nit: inline `rType.TypeName()` here and get rid of `typeName` variable.

**Review by easwars — APPROVED** [review]

**Comment by easwars** [comment]
@anicr7 : Thanks for doing this. Please let me know if you want me to merge this for you if you don't have write access to the repo.

File: xds/internal/testutils/testutils.go
Comment (dfawley) [inline xds/internal/testutils/testutils.go:43]: Should there be a `default: typS = typeName`?

Comment (anicr7) [reply xds/internal/testutils/testutils.go:47]: Done!

File: xds/internal/xdsclient/authority.go
Comment (anicr7) [reply xds/internal/xdsclient/authority.go]: Done

**Comment by anicr7** [comment]
> 

Thanks Easwar! Once Doug approves please feel free to merge, I don't have write access on the repo :).

**Review by easwars — APPROVED** [review]

**Pull request merged**
